neat_ml/tests/test_plotting.py

```diff
@@ -0,0 +1,379 @@
+import json
+from pathlib import Path
+from importlib import resources
+from typing import Callable, Generator, Any
+import matplotlib
+matplotlib.use("Agg")
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import pytest
+from matplotlib.testing.compare import compare_images
+import os
+
+from neat_ml.utils import lib_plotting as pmf
+from neat_ml.utils import figure_utils
+
+# TODO: enforce style consistency with ``black`` (issue #11)
+@pytest.fixture(scope="session")
+def synthetic_df() -> pd.DataFrame:
+    """Deterministic (seeded) composition/phase dataframe."""
+    rng = np.random.default_rng(7)
+    x = rng.uniform(0.0, 20.0, 30)
+    y = rng.uniform(0.0, 20.0, 30)
+    phase = (x + y > 20.0).astype(int)
+    return pd.DataFrame({"Sodium Citrate (wt%)": x,
+                         "PEO 8 kg/mol (wt%)": y,
+                         "Phase": phase})
+
+@pytest.fixture(scope="session")
+def baseline_dir() -> Generator[Any, Any, Any]:
+    """
+    Directory that stores the reference (expected) images.
+    """
+    ref = resources.files("neat_ml.tests") / "baseline"
+    with resources.as_file(ref) as path:
+        yield path
+
+def test_plot_gmm_decision_regions_visual_and_logic(
+    tmp_path: Path,
+    baseline_dir: Path,
+    synthetic_df: pd.DataFrame,
+):
+    fig, ax = plt.subplots(figsize=(6, 6), dpi=150)
+    gmm, labels, boundary = figure_utils.plot_gmm_decision_regions(
+        df=synthetic_df,
+        x_col=synthetic_df.columns[0],
+        y_col=synthetic_df.columns[1],
+        phase_col="Phase",
+        ax=ax,
+        xrange=[0, 20],
+        yrange=[0, 20],
+        n_components=2,
+        random_state=42,
+        boundary_color="red",
+        resolution=200,
+        decision_alpha=1,
+        plot_regions=True,
+        region_colors=["lightsteelblue", "aquamarine"],
+    )
+    assert labels.shape == (len(synthetic_df),)
+    assert boundary is None or boundary.shape[1] == 2
+    assert hasattr(gmm, "predict")
```

Comment:
I'm a bit confused--how is this test case different from one of the cases in `test_plotters_visual_and_logic()`? You may get more cases over there, but it certainly looks like there is overlap and I'm basically reading the same thing, and even the reference image `gmm_decision_regions.png` is already accounted for in two cases over there?

What is the basis for this design decision?